src/states/common/base.rs

```diff
@@ -304,13 +318,52 @@ pub trait Base: Display {
         &mut self,
         addr: SocketAddr,
         msg: Bytes,
-        token: Token,
-        _outbox: &mut dyn EventBox,
+        msg_token: Token,
+        outbox: &mut dyn EventBox,
     ) -> Transition {
-        let log_ident = LogIdent::new(self);
-        self.network_service_mut()
-            .send_message_to_next_target(msg, token, addr, log_ident);
-        Transition::Stay
+        let next_target = self.network_service_mut().target_failed(msg_token, addr);
```

Comment:
just to confirm that the `failing addr` could be picked as `next_target` as well.
I don't think it's a problem as the `send` shall just be scheduled later.

Reply:
That's right. Also If the number of previously failed attempts to that target is zero, the message will be sent to them immediately, otherwise with a delay.